### PR TITLE
Replace url connection with Unirest test framework

### PR DESCRIPTION
**Pull request by chamil321**

### CODE REVIEW

File: components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/contentaware/EchoServerConnectorListenerTestCase.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
```
Comment (Bhashinee) [inline components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/contentaware/EchoServerConnectorListenerTestCase.java]: Shouldn't this be 2018?

File: components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/contentaware/RequestResponseCreationTestCase.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        try {
+            serverConnectorFuture.sync();
+        } catch (InterruptedException e) {
+            log.error("Thread Interrupted while sleeping ", e);
+        }
+        httpServer = TestUtil.startHTTPServer(TestUtil.HTTP_SERVER_PORT, new EchoServerInitializer());
+    }
+
+    @Test
+    public void requestResponseCreationFromProcessorTestCase() {
```
Comment (shafreenAnfar) [inline components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/contentaware/RequestResponseCreationTestCase.java]: shall we rename this to testRequestResponseCreation

File: components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/contentaware/RequestResponseStreamingTestCase.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        try {
+            serverConnectorFuture.sync();
+        } catch (InterruptedException e) {
+            log.error("Thread Interrupted while sleeping ", e);
+        }
+        httpServer = TestUtil.startHTTPServer(TestUtil.HTTP_SERVER_PORT, new EchoServerInitializer());
+    }
+
+    @Test
+    public void requestResponseStreamingFromProcessorTestCase() {
```
Comment (shafreenAnfar) [inline components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/contentaware/RequestResponseStreamingTestCase.java]: Shall we rename this to testRequestResponseStreaming

File: components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/contentaware/RequestResponseTransformTestCase.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        try {
+            serverConnectorFuture.sync();
+        } catch (InterruptedException e) {
+            log.error("Thread Interrupted while sleeping ", e);
+        }
+        httpServer = TestUtil.startHTTPServer(TestUtil.HTTP_SERVER_PORT, new EchoServerInitializer());
+    }
+
+    @Test
+    public void requestResponseTransformFromProcessorTestCase() {
```
Comment (shafreenAnfar) [inline components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/contentaware/RequestResponseTransformTestCase.java]: Shall rename this to testRequestResponseTranformation

**Pull request merged**